test/issue_746_helper.c

```diff
@@ -0,0 +1,383 @@
+/*
```

Comment:
I suggest giving this file a more descriptive name perhaps, so that people don't have to refer back to github?

Reply:
Keep the initial example/passthrough name? I guess it could be used for other tests as well.

Reply:
Name suggestions?

Reply:
Maybe `release_unlink_race`?